hax/hax/ha/events.py

```diff
@@ -145,3 +152,78 @@ def ack(self):
            that message using ack method
         """
         self.consumer.ack()
+
+
+class EventProducer():
+    """
+    Module responsible for publishing the event to message bus
+    """
+
+    ADMIN_ID = "hare_admin"
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        """
+        Static method to fetch the current instance.
+        Performs initialization related to common database and message bus
+        """
+        if not EventProducer.__instance:
+            EventProducer()
+        return EventProducer.__instance
+
+    def __init__(self):
+        logging.debug('Inside EventProducer')
+
+        if EventProducer.__instance is None:
+            EventProducer.__instance = self
+        else:
+            raise Exception("EventProducer is singleton class,"
+                            " use EventProducer.get_instance().")
+
+        message_type = FAULT_TOLERANCE_KEYS.HARE_HA_MESSAGE_TYPE.value
+        self._register_producer(message_type)
+
+        self.producer = MessageProducer(producer_id=HEALTH_EVENT_SOURCES.HARE,
+                                        message_type=message_type)
+
+    def _register_producer(self, message_type,
+                           partitions: int = 1):
+        util: ConsulUtil = ConsulUtil()
+        configpath = util.get_configpath()
+        Conf.load('cortx_conf', configpath)
+
+        message_server_endpoints = Conf.get('cortx_conf',
+                                            'cortx>external>kafka>endpoints')
+        MessageBus.init(message_server_endpoints)
+
+        admin = MessageBusAdmin(admin_id=self.ADMIN_ID)
+        try:
+            if message_type not in admin.list_message_types():
+                admin.register_message_type(message_types=[message_type],
+                                            partitions=partitions)
+        except Exception as e:
+            if "TOPIC_ALREADY_EXISTS" not in str(e):
+                raise(e)
+
+    def create_node_event(self, res_name, health_status, generation_id):
+        self.payload = {
+            HealthAttr.SOURCE.value: HEALTH_EVENT_SOURCES.HARE.value,
+            HealthAttr.CLUSTER_ID.value: NOT_DEFINED,
+            HealthAttr.SITE_ID.value: NOT_DEFINED,
+            HealthAttr.RACK_ID.value: NOT_DEFINED,
+            HealthAttr.STORAGESET_ID.value: NOT_DEFINED,
+            HealthAttr.NODE_ID.value: res_name,
+            HealthAttr.RESOURCE_TYPE.value: 'node',
+            HealthAttr.RESOURCE_ID.value: res_name,
+            HealthAttr.RESOURCE_STATUS.value: health_status,
+            HealthAttr.SPECIFIC_INFO.value: {}
+            }
+
+        self.event = HealthEvent(**self.payload)
+        self.event.set_specific_info({"generation_id": generation_id})
```

Comment:
How this `generation_id` is generated?

Reply:
generation_id will be pod name in case of Node health event according to HLD.

Reply:
Okay, please add a small comment for generation id.